src/wazuh_db/wdb_agent.c

```diff
@@ -11,269 +11,295 @@
+    [SQL_UPDATE_FIM_OFFSET] = "global sql UPDATE agent SET fim_offset = %lu WHERE id = %d;",
+    [SQL_UPDATE_REG_OFFSET] = "globL sql UPDATE agent SET reg_offset = %lu WHERE id = %d;",
+    [SQL_SELECT_AGENT_STATUS] = "global sql SELECT status FROM agent WHERE id = %d;",
+    [SQL_UPDATE_AGENT_STATUS] = "global sql UPDATE agent SET status = %Q WHERE id = %d;",
+    [SQL_UPDATE_AGENT_GROUP] = "global sql UPDATE agent SET `group` = %Q WHERE id = %d;",
+    [SQL_FIND_GROUP] = "global sql SELECT id FROM `group` WHERE name = %Q;",
+    [SQL_INSERT_AGENT_GROUP] = "global sql INSERT INTO `group` (name) VALUES(%Q);",
+    [SQL_INSERT_AGENT_BELONG] = "global sql INSERT INTO belongs (id_group, id_agent) VALUES(%d, %d);",
+    [SQL_DELETE_AGENT_BELONG] = "global sql DELETE FROM belongs WHERE id_agent = %d",
+    [SQL_DELETE_GROUP_BELONG] = "global sql DELETE FROM belongs WHERE id_group = (SELECT id FROM 'group' WHERE name = %Q );", 
+    [SQL_DELETE_GROUP] = "global sql DELETE FROM `group` WHERE name = %Q;",
+    [SQL_SELECT_GROUPS] = "global sql SELECT name FROM `group`;"
+ };
 
 /* Insert agent. It opens and closes the DB. Returns 0 on success or -1 on error. */
 int wdb_insert_agent(int id, const char *name, const char *ip, const char *register_ip, const char *key, const char *group, int keep_date) {
     int result = 0;
-    sqlite3_stmt *stmt;
-    const char * sql = SQL_INSERT_AGENT;
-    time_t date;
-
-    if (wdb_open_global() < 0)
-        return -1;
+    time_t date = 0;
+    char wdbquery[OS_BUFFER_SIZE] = "";
+    char wdboutput[OS_BUFFER_SIZE] = "";
+    int wdb_sock = -1;
 
     if(keep_date) {
         date = get_agent_date_added(id);
     } else {
         time(&date);
     }
 
-    if (wdb_prepare(wdb_global, sql, -1, &stmt, NULL)) {
-        mdebug1("SQLite: %s", sqlite3_errmsg(wdb_global));
-        return -1;
-    }
+    sqlite3_snprintf(sizeof(wdbquery), wdbquery,  global_db_queries[SQL_INSERT_AGENT], id, name, ip, register_ip, key, date, group);
+    result = wdbc_query_ex(&wdb_sock, wdbquery, wdboutput, sizeof(wdboutput));
 
-    sqlite3_bind_int(stmt, 1, id);
-    sqlite3_bind_text(stmt, 2, name, -1, NULL);
-
-    if (ip){
-        sqlite3_bind_text(stmt, 3, ip, -1, NULL);
-    }
-    else {
-        sqlite3_bind_null(stmt, 3);
-    }
-
-    if(register_ip){
-        sqlite3_bind_text(stmt, 4, register_ip, -1, NULL);
-    }
-    else{
-        sqlite3_bind_null(stmt, 4);
+    switch (result){
+        case OS_SUCCESS:
+            result = wdb_create_agent_db(id, name);
+            break;
+        case OS_INVALID:
+            mdebug1("GLobal DB Error in the response from socket");
+            mdebug2("Global DB SQL query: %s", wdbquery);
+            break;
+        default:
+            mdebug1("GLobal DB Cannot execute SQL query; err database %s/%s.db", WDB2_DIR, WDB2_GLOB_NAME);
+            mdebug2("Global DB SQL query: %s", wdbquery);
+            break;
```

Comment:
The `break` is not necessary in the `default` case

Reply:
Replaced by `result = OS_INVALID;`